stat/card/card_test.go

```diff
@@ -195,6 +196,96 @@ func TestResetCounters(t *testing.T) {
 	}
 }
 
+type counterEncoder interface {
+	counter
+	encoding.BinaryMarshaler
+	encoding.BinaryUnmarshaler
+}
+
+var counterEncoderTests = []struct {
+	name     string
+	count    int
+	src, dst func() counterEncoder
+}{
+	{
+		name: "HyperLogLog32-4-4-FNV-1a", count: 1e3,
+		src: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog32(4, fnv.New32a())) },
+		dst: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog32(4, fnv.New32a())) },
+	},
+	{
+		name: "HyperLogLog32-4-8-FNV-1a", count: 1e3,
+		src: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog32(4, fnv.New32a())) },
+		dst: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog32(8, fnv.New32a())) },
+	},
+	{
+		name: "HyperLogLog32-8-4-FNV-1a", count: 1e3,
+		src: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog32(8, fnv.New32a())) },
+		dst: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog32(4, fnv.New32a())) },
+	},
+	{
+		name: "HyperLogLog64-4-4-FNV-1a", count: 1e3,
+		src: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog64(4, fnv.New64a())) },
+		dst: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog64(4, fnv.New64a())) },
+	},
+	{
+		name: "HyperLogLog64-4-8-FNV-1a", count: 1e3,
+		src: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog64(4, fnv.New64a())) },
+		dst: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog64(8, fnv.New64a())) },
+	},
+	{
+		name: "HyperLogLog64-8-4-FNV-1a", count: 1e3,
+		src: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog64(8, fnv.New64a())) },
+		dst: func() counterEncoder { return mustCounterEncoder(NewHyperLogLog64(4, fnv.New64a())) },
+	},
+}
+
+func mustCounterEncoder(c counterEncoder, err error) counterEncoder {
+	if err != nil {
+		panic(fmt.Sprintf("bad test: %v", err))
```

Comment:
Suggest not using panics.
You can use https://godoc.org/testing#T.Fatal

Reply:
Not possible here; these calls happen in package-level vars where there is no `*testing.T` available. The errors returned by the constructors are solely dependent on the precision parameter and are essentially a compile time error, so panics are fine here.